src/filters/sse.rs

```diff
@@ -52,119 +53,126 @@ use futures::{future, Stream, TryStream, TryStreamExt};
 use http::header::{HeaderValue, CACHE_CONTROL, CONTENT_TYPE};
 use hyper::Body;
 use pin_project::pin_project;
-use serde::Serialize;
-use serde_json;
+use serde_json::{self, Result as SerdeResult};
 use tokio::time::{self, Delay};
 
-use self::sealed::{
-    BoxedServerSentEvent, EitherServerSentEvent, SseError, SseField, SseFormat, SseWrapper,
-};
+use self::sealed::SseError;
 use super::header;
 use crate::filter::One;
 use crate::reply::Response;
 use crate::{Filter, Rejection, Reply};
 
-/// Server-sent event message
-pub trait ServerSentEvent: SseFormat + Sized + Send + Sync + 'static {
-    /// Convert to either A
-    fn into_a<B>(self) -> EitherServerSentEvent<Self, B> {
-        EitherServerSentEvent::A(self)
+// Server-sent event data type
+#[derive(Debug)]
+enum DataType {
+    Text(String),
+    Json(SerdeResult<String>),
+}
+
+/// Server-sent event
+#[derive(Default, Debug)]
+pub struct Event {
+    name: Option<String>,
+    id: Option<String>,
+    data: Option<DataType>,
+    event: Option<String>,
+    comment: Option<String>,
+    retry: Option<Duration>,
+}
+
+impl Event {
+    /// Set Server-sent event data
+    /// data field(s) ("data:<content>")
+    pub fn set_data<T: Display>(mut self, data: Option<T>) -> Event {
+        self.data = data.map(|d| DataType::Text(d.to_string()));
+        self
     }
 
-    /// Convert to either B
-    fn into_b<A>(self) -> EitherServerSentEvent<A, Self> {
-        EitherServerSentEvent::B(self)
+    /// Set Server-sent event data
+    /// data field(s) ("data:<content>")
+    pub fn set_json_data<T: Serialize>(mut self, data: Option<T>) -> Event {
```

Comment:
I'm not sure about swallowing the error in here. What if this returned a Result instead?

Reply:
yeah, i was not sure either, opted for swallowing because of the previous API and the issue with `self` being consumed on case of Error. updated!